### ISSUE TITLE
[AMP] Bad decomp for softmax

### ISSUE THREAD

**Issue by anijain2305** [pytorch/torchdynamo#1619]
Repro

~~~
from math import inf

import torch
import torch.fx as fx
from torch import device
from torch import tensor

import torchdynamo
from torchdynamo.debug_utils import run_fwd_maybe_bwd
from torchdynamo.debug_utils import same_two_models
from torchdynamo.testing import rand_strided

args = [
    ((1, 12, 128, 128), (196608, 16384, 128, 1), torch.float16, "cuda", True),
    ((1, 12, 128, 128), (128, 0, 0, 1), torch.bool, "cuda", False),
    ((), (), torch.float32, "cpu", False),
]
args = [
    rand_strided(sh, st, dt, dev).requires_grad_(rg) for (sh, st, dt, dev, rg) in args
]


from torch.nn import *


class Repro(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, matmul, expand_as, tensor):
        masked_fill = matmul.masked_fill(expand_as, tensor)
        softmax = torch.nn.functional.softmax(masked_fill, dim=-1)
        return (softmax,)


mod = Repro().cuda()
opt_mod = torchdynamo.optimize("aot_inductor_debug")(mod)


with torch.cuda.amp.autocast(enabled=True):
    ref = run_fwd_maybe_bwd(mod, args)
    res = run_fwd_maybe_bwd(opt_mod, args)


~~~
